packages/ui/src/hooks/use-social.ts

```diff
@@ -0,0 +1,184 @@
+import { useEffect, useCallback, useContext } from 'react';
+import { useParams } from 'react-router-dom';
+
+import { invokeSocialSignIn, signInWithSocial } from '@/apis/social';
+import { generateRandomString, parseQueryParameters } from '@/utils';
+
+import PageContext from './page-context';
+import useApi from './use-api';
+
+/**
+ * Social Connector State Utility Methods
+ * @param state
+ * @param state.uunid - unique id
+ * @param state.platform - platform
+ * @param state.callbackUriScheme - callback uri scheme
+ */
+
+type State = {
+  uunid: string;
+  platform: 'web' | 'ios' | 'android';
+  callbackUriScheme?: string;
+};
+
+const storageKeyPrefix = 'social_auth_state';
+
+export const generateState = () => {
+  const uunid = generateRandomString();
+  const platform = logtoNativeSdk?.platform ?? 'web';
+  const callbackUriScheme = logtoNativeSdk?.callbackUriScheme;
+
+  const state: State = { uunid, platform, callbackUriScheme };
+
+  return Buffer.from(JSON.stringify(state)).toString('base64');
```

Comment:
can we use native `btoa` function?

Reply:
<img width="723" alt="image" src="https://user-images.githubusercontent.com/36393111/163939163-d479368c-11c5-4fc8-b7d0-28833c07e4d6.png">

technically browser is fine? I'll update then

Reply:
updated